doc/reference/actor.md

```diff
@@ -2,755 +2,738 @@
 ```
-(send-message ...)
-```
-:::
 
-Please document me!
+Structure for actor errors.
 
-### send-message/timeout
-::: tip usage
-```
-(send-message/timeout ...)
+### raise-actor-error
+```scheme
+(raise-actor-error where what . irritants)
 ```
-:::
 
-Please document me!
+Raises an actor error.
 
-### &lt;&lt;
-::: tip usage
-```
-(<< ...)
+### default-reply-timeout
+```scheme
+(default-reply-timeout)
 ```
-:::
 
-Please document me!
+The default reply timeout in seconds; initial value is 5s.
 
-### &lt;-
-::: tip usage
+### set-default-reply-timeout!
+```scheme
+(set-default-reply-timeout! timeo)
 ```
-(<- ...)
-```
-:::
+Sets the default reply timeout.
 
-Please document me!
 
-### @message
-```
-(defsyntax @message)
-(defsyntax @@message)
+## Handles and References
+
+### make-handle
+```scheme
+(handle proxy ref)
 ```
 
-Please document me!
+Creates a handle for sending messages to an actor through a proxy.
+- `proxy` is the actor who will receive the messages, a thread.
+- `ref` is a `reference` to the actor being proxied; in general it is
+  something the proxy can interpret; see `referece` below.
 
-### @value
-```
-(defsyntax @value)
-(defsyntax @@value)
+### handle?
+```scheme
+(handle? obj)
 ```
 
-Please document me!
+Predicate for handles.
 
-### @source
-```
-(defsyntax @source)
-(defsyntax @@source)
+
+### handle-proxy
+```scheme
+(handle-proxy h)
 ```
 
-Please document me!
+Returns the proxy in the handle.
 
-### @dest
-```
-(defsyntax @dest)
-(defsyntax @@dest)
+### handle-ref
+```scheme
+(handle-ref h)
 ```
 
-Please document me!
+Returns the actor reference in the handle.
 
-### @options
-```
-(defsyntax @options)
-(defsyntax @@options)
+### reference
+```scheme
+(defmessage reference (server id))
 ```
 
-Please document me!
+References for actors in the ensemble.
+- `server` is the server identifier where the actor resides.
+- `id` is the actor identifier, generally a symbol for references by name.
 
-### !
-```
-(defsyntax !)
+
+### reference->handle
+```scheme
+(reference->handle ref (srv (current-actor-server)))
 ```
 
-Please document me!
+Creates a handle from a reference, using by default the current actor
+server as the proxy.
 
 
 ## Protocols
 
-### defproto
-::: tip usage
+### defmessage
+```scheme
+(defmessage id (field-name ...) struct-options ...)
 ```
-(defproto ...)
-```
-:::
 
-Please document me!
+Macro to define message types that can be efficiently marshalled as
+protocol messages.
 
-### defproto-default-type
-::: tip usage
-```
-(defproto-default-type ...)
-```
-:::
+The structure is final and transparent, and it is registered in the
+message type registry where the unmarshaller can find it.
 
-Please document me!
+::: note
+Messages _must_ be acyclic; if you want to send cyclic data you
+can use a normal struct, but be aware that such structs will be
+serialized/deserialized with the raw gambit serializer and carry the
+whole type descriptor (including methods) with them.
```

Comment:
It does work remotely and there is no short circuit.

What it entails: you need to understand how the serializer works with structs: it will put the type descriptor in, and that will include the method tables etc.

I don't think we need to document this in detail, it is _too much_ detail.